File: DP/김상효/boj_11055_가장 큰 증가하는 부분 수열.swift
//
//  main.swift
//  Algorithm
//
//  Created by sanghyo on 2023/03/28.
//

let n = Int(readLine()!)!
let arr = readLine()!.split(separator: " ").map { Int($0)! }
var dp: [Int] = Array(repeating: 0, count: n)
dp[0] = arr[0]

for i in 1..<n {
    dp[i] = arr[i]
    for j in stride(from: i-1, through: 0, by: -1) {
        if arr[j] < arr[i] {
            dp[i] = max(dp[i], dp[j] + arr[i])
        }
    }
}

print(dp.max()!)

File: DP/박성수/boj_11055_가장큰증가하는부분수열.py
N = int(input())
arr = list(map(int, input().split()))

dp = arr[:]

for i in range(N):
    for j in range(i):
        if arr[j] < arr[i]:
            dp[i] = max(dp[i], dp[j] + arr[i])

print(*dp)
print(max(dp))



